Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            HVCondition hvCondition = new HVCondition();
+
+            Guid id;
+            if (Guid.TryParse(fhircondition.Id, out id))
+            {
+                hvCondition.Key = new ThingKey(id);
+            }
+
+            Guid version;
+            if (fhircondition.Meta != null && fhircondition.Meta.VersionId != null && Guid.TryParse(fhircondition.Meta.VersionId, out version))
+            {
+                hvCondition.Key.VersionStamp = version;
+            }
+
+            hvCondition.StopReason = fhircondition.GetStringExtension(HealthVaultVocabularies.ConditionStopReason);
+            hvCondition.CommonData.Source = fhircondition.GetStringExtension(HealthVaultVocabularies.ConditionSource);
+            hvCondition.OnsetDate = GetHealthVaultApproximateTimeFromEffectiveDate(fhircondition.Onset);
+            hvCondition.StopDate = GetHealthVaultApproximateTimeFromEffectiveDate(fhircondition.Abatement);
+
+            if (fhircondition.ClinicalStatus.HasValue)
+            {
+                hvCondition.Status =  new ItemTypes.CodableValue(fhircondition.ClinicalStatus.Value.ToString());
+                hvCondition.Status.Add(new ItemTypes.CodedValue(fhircondition.ClinicalStatus.Value.ToString(), FhirCategories.Hl7Condition,HealthVaultVocabularies.Fhir, ""));
+            }
+            else
+            {
+                hvCondition.Status = new ItemTypes.CodableValue(fhircondition.GetStringExtension(HealthVaultVocabularies.ConditionOccurenceExentsionName));
+                hvCondition.Status.Add(new ItemTypes.CodedValue(fhircondition.GetStringExtension
+                    (HealthVaultVocabularies.ConditionOccurenceExentsionName), HealthVaultVocabularies.ConditionOccurence, HealthVaultVocabularies.Wc, "1"));
+            }
+
+            if (fhircondition.Code != null && fhircondition.Code.Coding != null)
+            {
+                foreach (var code in fhircondition.Code.Coding)
+                {
+                    var value = code.Code.Split(':');
+                    var vocabCode = value.Length == 2 ? value[1] : null;
+                    var vocab = value.Length == 2 ? value[0] : HealthVaultVocabularies.Fhir;
+
+                    hvCondition.SetConditionName(
+                            code.Display,
+                            vocabCode,
+                            vocab,
+                            code.System,
+                            code.Version);                    
+                }
+            }
+
+            if (fhircondition.Note != null)
+            {
+                foreach (Annotation annotation in fhircondition.Note)
+                {
+                    hvCondition.CommonData.Note += annotation.Text;
+                }                
+            }
+            
+            return hvCondition;
+        }
+        private static ItemTypes.ApproximateDateTime GetHealthVaultApproximateTimeFromEffectiveDate(Element fhirElement)
```

Comment:
>GetHealthVaultApproximateTimeFromEffectiveDate [](start = 53, length = 46)

This function doesn't belong here, it looks too generic to be only related to Condition, please move out